app/repository/users_repository.py

```diff
@@ -0,0 +1,45 @@
+import uuid
+from collections.abc import Sequence
+from typing import Any
+
+from sqlmodel import select
+from sqlmodel.ext.asyncio.session import AsyncSession
+
+from app.models.user import User, UserCreate
+from app.utilities.exceptions import NotFoundException, NotUniqueException
+
+
+class UsersRepository:
+    def __init__(self, session: AsyncSession) -> None:
+        self.session = session
+
+    async def _check_unique_attr(self, attr: Any, value: Any) -> None:
+        statement = select(getattr(User, attr)).where(getattr(User, attr) == value)
+        exists = (await self.session.exec(statement)).first() is not None
+        if exists:
+            raise NotUniqueException(item=attr.capitalize())
```

Comment:
Me parecería más legible si está escrito:
- evitaría los doble negado (not None) que a veces confunde

```suggestion
        user_attr = getattr(User, attr)
        statement = select(user_attr).where(user_attr == value)
        attr_exist = (await self.session.exec(statement)).first()
        if attr_exist:
            raise NotUniqueException(item=attr.capitalize())
```

Reply:
Procedo a realizar la corrección solicitada